binding/android/leopard/src/main/java/ai/picovoice/leopard/Leopard.java

```diff
@@ -0,0 +1,166 @@
+/*
+    Copyright 2022 Picovoice Inc.
+    You may not use this file except in compliance with the license. A copy of the license is
+    located in the "LICENSE" file accompanying this source.
+    Unless required by applicable law or agreed to in writing, software distributed under the
+    License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+    express or implied. See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+
+package ai.picovoice.leopard;
+
+import android.content.Context;
+import android.content.res.Resources;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.regex.Pattern;
+
+/**
+ * Android binding for Leopard Speech-to-Text engine.
+ */
+public class Leopard {
+
+    private static String defaultModelPath;
+
+    static {
+        System.loadLibrary("pv_leopard");
+    }
+
+    private final long handle;
+
+    /**
+     * Constructor.
+     *
+     * @param accessKey AccessKey obtained from Picovoice Console
+     * @param modelPath Absolute path to the file containing Leopard model parameters.
+     * @throws LeopardException if there is an error while initializing Leopard.
+     */
+    public Leopard(String accessKey, String modelPath) throws LeopardException {
+        handle = init(accessKey, modelPath);
+    }
+
+    private static String extractResource(Context context, InputStream srcFileStream, String dstFilename) throws IOException {
+        InputStream is = new BufferedInputStream(srcFileStream, 256);
+        OutputStream os = new BufferedOutputStream(context.openFileOutput(dstFilename, Context.MODE_PRIVATE), 256);
+        int r;
+        while ((r = is.read()) != -1) {
```

Comment:
can we read more than one byte at a time?

Reply:
Not sure, this is just how we do it for all the other bindings so I figured I'd just leave it as since we know it works well.

Reply:
I think this is one of those situations that a mistake propagated over time. Let's fix it here and make sure the fix goes into Cheetah as well. Looking at the [doc](https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#read()) this call is reading 1 byte at a time. Let's try to read more ... like up to 1k or something? It will be much faster. We are reading 20 MB 1 byte at a time it will take a while.

Reply:
Okay, I'll see what I can do

Reply:
So the reason it's like this is because there isn't actually that much better a way to do it with the Android versions we support. Doing it it better would require us to use the `nio` library which is only supported in Android 6+. Both the input and output streams are buffered, so I'd be willing to bet it doesn't make all that much a spectacle difference.

Reply:
This should read 256 bytes at a time because of the size of the buffered streams. I believe you can increase the buffer stream sizes from 256 to 1024 if you want to read 1K at a time.